### PR TITLE
feat: include optional ForkId in Discovery

### LINKED ISSUE #856: Include optional ForkId when a new NodeRecord is discovered
### Describe the feature

We have two sources in discovery, dns and discv2, DNS can include the ForkId directly when the node is discovered:

https://github.com/paradigmxyz/reth/blob/e6ca4c56c6a314e5553656e5b62be3bca6801b85/crates/net/network/src/discovery.rs#L127

This should be propagated to the peersset via this message:

https://github.com/paradigmxyz/reth/blob/e6ca4c56c6a314e5553656e5b62be3bca6801b85/crates/net/network/src/discovery.rs#L134

## TODO

* include `Option<ForkId>` in `Discovered` event, refactor variant to named fields
* Integrate Option ForkId in `PeersManager`:

https://github.com/paradigmxyz/reth/blob/e6ca4c56c6a314e5553656e5b62be3bca6801b85/crates/net/network/src/peers/manager.rs#L428

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TechieBoy**
Closes #856

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]
Looks good to me!

In the future would you kindly comment on the issue when you start working on it? It lets us assign you to the issue (GitHub won't allow us to do it without the comment) which gives others a better sense of what is being worked on and what is up for grabs :smile:

**Comment by TechieBoy** [comment]
> In the future would you kindly comment on the issue

Sure! Will keep in mind

File: crates/net/network/src/peers/manager.rs [hunk 1]
```diff
@@ -424,9 +424,12 @@ impl PeersManager {
...
     ///
-    /// If the peer already exists, then the address and kind will be updated.
-    pub(crate) fn add_peer(&mut self, peer_id: PeerId, addr: SocketAddr) {
-        self.add_peer_kind(peer_id, PeerKind::Basic, addr)
+    /// If the peer already exists, then the address, kind and fork_id will be updated.
+    pub(crate) fn add_peer(&mut self, peer_id: PeerId, addr: SocketAddr, fork_id: Option<ForkId>) {
+        self.add_peer_kind(peer_id, PeerKind::Basic, addr);
+        if let Some(fork_id) = fork_id {
+            self.set_discovered_fork_id(peer_id, fork_id);
+        }
```
Comment (mattsse) [inline crates/net/network/src/peers/manager.rs:432]: these are two lookups now, one forst insert Peer then set forkId,

I think we can unify this by extending `add_peer_kind` with `Option<ForkId>` and initialize a `Peer` with that.

**Review by mattsse — CHANGES_REQUESTED** [review]
ty!

smol request to optimize new peer inserts

**Review by mattsse — APPROVED** [review]
lgtm!

**Comment by mattsse** [comment]
@TechieBoy just realized we're missing a validate check if the fork id is present: https://github.com/paradigmxyz/reth/issues/864

do you want to tackle this next, should just be a simple if check once this Pr is merged

**Comment by TechieBoy** [comment]
> do you want to tackle this next, should just be a simple if check once this Pr is merged

Yup, will tackle it next

**Pull request merged**